script.sh

```diff
@@ -208,14 +209,40 @@ RUN echo "deb-src ${TRAVIS_DEBIAN_MIRROR} experimental main" >> /etc/apt/sources
 EOF
 fi
 
+if [ "$(echo ${TRAVIS_EXTRA_REPOSITORY} | cut -c1-6)" = "https:" ]
+then
+	EXTRA_PACKAGES="apt-transport-https"
+fi
+
+if [ -n "${TRAVIS_EXTRA_REPOSITORY_GPG_URL}" ]
```

Comment:
I'd also like to see some (brief!) documentation addition to ``index.html`` — no need for it to be perfect, it just has to quickly go through what these new variables do and their defaults.